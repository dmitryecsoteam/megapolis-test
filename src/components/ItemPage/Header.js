import React, { Fragment } from 'react';
import PropTypes from 'prop-types';


const Header = ({ items, match, history, deleteItem }) => {

    // Находим значение id из объекта match, который предоставляет роутер
    const id = parseInt(match.params.id);

    // Удаляем айтем и перенаправляем на страницу со списком
    const onDeleteClick = () => {
        deleteItem(id);
        history.push('/items');
    }

    const header = <div className='header__container'>
        <h1 className='header__title'>Задача №{id}</h1>
        <button
            className='button button--small button--red'
            onClick={onDeleteClick}
        >
            Удалить
    </button>
    </div>;

    // Отрендерить header, только если данные загружены из апи
    return (
        <Fragment>
            {items[0] !== 'INIT' && header}
        </Fragment>

    );
}

Header.propTypes = {
    items: PropTypes.array.isRequired,
    history: PropTypes.object.isRequired,
    match: PropTypes.object.isRequired,
    deleteItem: PropTypes.func.isRequired
}

export default Header;