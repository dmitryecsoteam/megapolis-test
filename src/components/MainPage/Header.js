import React from 'react';
import Popup from 'reactjs-popup';

class Header extends React.Component {
    state = {
        title: ''
    }

    onTitleChange = ({ target }) => {
        this.setState({
            title: target.value
        });
    }

    onFormSubmit = (e, close) => {
        e.preventDefault();

        this.props.addItem(this.state.title, Math.floor(Math.random() * 1000));

        this.setState({
            title: ''
        });

        close();
    }

    render() {
        return (
            <div className='header__container'>
                <h1 className='header__title'>Список задач</h1>
                <Popup
                    trigger={<button className='header__add-button'>Добавить</button>}
                    modal
                    closeOnDocumentClick
                >
                    {close => (
                        <div className='add-popup__container'>
                            <form onSubmit={(e) => this.onFormSubmit(e, close)}>
                                <input
                                    type="text"
                                    value={this.state.title}
                                    onChange={this.onTitleChange}
                                />
                                <button>Создать</button>
                            </form>
                        </div>
                    )}
                </Popup>
            </div>
        )
    }

}

export default Header;