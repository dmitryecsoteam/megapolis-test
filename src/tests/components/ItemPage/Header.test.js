import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../../components/ItemPage/Header';

let wrapper;
const history = {
    push: jest.fn()
};
const match = {
    params: { id: 1 }
};
const deleteItem = jest.fn();

beforeEach(() => {
    wrapper = shallow(<Header items={['INIT']} history={history} match={match} deleteItem={deleteItem}/>);
});

test('should not render Header if items array is "INIT"', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should render Header', () => {
    wrapper.setProps({
        items: [{ id: 1, title: 'first' }]
    });
    expect(wrapper).toMatchSnapshot();
});

test('should delete item on button click', () => {
    wrapper.setProps({
        items: [{ id: 1, title: 'first' }]
    });
    wrapper.find('button').simulate('click');
    expect(deleteItem).toHaveBeenCalledWith(1);
    expect(history.push).toHaveBeenCalledWith('/items');
});