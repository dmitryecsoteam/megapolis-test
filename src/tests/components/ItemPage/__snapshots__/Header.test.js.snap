// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not render Header if items array is "INIT" 1`] = `<Fragment />`;

exports[`should render Header 1`] = `
<Fragment>
  <div
    className="header__container"
  >
    <h1
      className="header__title"
    >
      Задача №
      1
    </h1>
    <button
      className="button button--small button--red"
      onClick={[Function]}
    >
      Удалить
    </button>
  </div>
</Fragment>
`;
